crates/ibc/src/core/context/acknowledgement.rs

```diff
@@ -110,3 +110,175 @@ where
 
     Ok(())
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use rstest::*;
+
+    use crate::core::ics03_connection::connection::Counterparty as ConnectionCounterparty;
+    use crate::core::ics03_connection::connection::State as ConnectionState;
+    use crate::core::ics04_channel::channel::Counterparty;
+    use crate::core::ics04_channel::channel::State;
+    use crate::core::ics04_channel::Version;
+    use crate::core::ics24_host::identifier::ChannelId;
+    use crate::core::ics24_host::identifier::PortId;
+    use crate::{
+        applications::transfer::MODULE_ID_STR,
+        core::{
+            ics03_connection::{connection::ConnectionEnd, version::get_compatible_versions},
+            ics04_channel::{
+                channel::ChannelEnd, commitment::PacketCommitment,
+                msgs::acknowledgement::test_util::get_dummy_raw_msg_acknowledgement,
+            },
+            ics24_host::identifier::{ClientId, ConnectionId},
+        },
+        mock::context::MockContext,
+        test_utils::DummyTransferModule,
+        timestamp::ZERO_DURATION,
+        Height,
+    };
+
+    struct Fixture {
+        ctx: MockContext,
+        module_id: ModuleId,
+        msg: MsgAcknowledgement,
+        packet_commitment: PacketCommitment,
+        conn_end_on_a: ConnectionEnd,
+        chan_end_on_a_ordered: ChannelEnd,
+        chan_end_on_a_unordered: ChannelEnd,
+    }
+
+    #[fixture]
+    fn fixture() -> Fixture {
```

Comment:
I noticed if we turn the `fixture()` into a regular function without the `#[fixture]` attribute and then call it inside each test using the `#[test]` attribute, it still works. So, what's the point of using `rstest` here? Am I overlooking something?

Reply:
That's right, as [documented](https://docs.rs/rstest/latest/rstest/attr.rstest.html#injecting-fixtures). This is basically equivalent to `setup()` functions in [javascript testing frameworks](https://jestjs.io/docs/setup-teardown).

Since we currently only use this feature from `rstest`, you're right that we could do without (I wasn't sure going in if I'd need more features). Note also that there's a lot of copy/paste currently in our unit tests; I suspect that some  features in `rstest` could help with that in the future, and I don't think it hurts to keep it for now. 

WDYT?

Reply:
I attempted to fit it to the `ics03_connection` tests, but honestly, it couldn't do much better than what we already have, and at its best, it accomplished something we could achieve by refactoring our test functions.
Also, while trying to avoid rewriting some test scenarios by applying the `#[case]` attribute, I found out it is on Rust edition 2018 and conflicts with edition 2021
By the way, it may be helpful in some cases for reducing code duplication (which I also dislike). But, in places like this, I would prefer not to add another dep and use built-in Rust features. Ofc, if we find the `rstest` providing further value, at any point in the future, we can always import it there.

Reply:
Sure, I'm okay with getting rid of it. Feel free to not use it with the connection tests. I'd create a low-priority issue to remove it though, and take care of it when we figure out our testing strategy.